### DISCUSSION TITLE
When will api-platform 2.7 be available?

### DISCUSSION THREAD

**Discussion by NotionCommotion** [api-platform/api-platform#2020] [Q&A] [answered]
https://github.com/api-platform/core/blob/main/CHANGELOG.md#270 describes `2.7.0` but it is just an alias to `2.7.x-dev` which is an alias to `dev-main`.  My workaround is to change `composer.json` as follows but I would rather not do.  Thank you

```
{
    ...
    "minimum-stability": "dev",
    "prefer-stable": true,
    "require": {
        ...
        "api-platform/core": "2.7.x-dev",
    },
}
```

**Comment by alanpoulain** [comment, marked as answer]
We don't give dates for API Platform releases. It is: "when it's ready" :wink:

**Reply by NotionCommotion** [reply]
@alanpoulain Understood Alan.  Throw out a date out and everyone's saying "why isn't it done, you said so...".  In the mean time, wish to use  PHP8 attributes with Doctrine and will therefore change api/composer.json to use the dev api-platform package.  When doing so previously, I recall needing to change how security is set up and a few other items.  Would it be of any help for me to provide a detailed list of what needs to be changed?

**Reply by alanpoulain** [reply]
There are a lot of changes this time, but we have provided a Rector tool to do it automatically: https://github.com/api-platform/core/pull/4334.
The command is `bin/console api:rector:upgrade`.

**Reply by NotionCommotion** [reply]
@alanpoulain Thanks, I was unaware of this Rector.  I actually previously used [DoctrineAnnotationClassToAttributeRector](https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#doctrineannotationclasstoattributerector) and [AnnotationToAttributeRector](https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#php80) to likely accomplish much of the same.  Do you know if there are other changes made by https://github.com/api-platform/core/pull/4334?  Also, it is my understanding api-platform 2.6 doesn't support Doctrine annotations and only 2.7, agree?

**Reply by NotionCommotion** [reply]
Hello again alanpoulain,  Figured out how to use api:rector:upgrade,  Thank you

```
docker-compose exec php \
	composer require api-platform/core:2.7.x-dev

# When first attempting to execute api:rector:upgrade, I was incorrectly instructed to execute the following
#docker-compose exec php \
#	composer require --dev rector/rector-src

# But do this instead
docker-compose exec php \
	composer require --dev rector/rector

# And no missing Node class error :)
docker-compose exec php \
	bin/console api:rector:upgrade -d src

```

**Reply by deivid11** [reply]
Command changed to api:upgrade-resource  https://github.com/api-platform/core/pull/4671
